daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -11,34 +15,46 @@ module DA.Daml.Helper.Ledger (
     runLedgerAllocateParties,
     runLedgerUploadDar,
     runLedgerFetchDar,
-    runLedgerNavigator
+    runLedgerNavigator,
     ) where
 
-import Control.Exception.Safe (SomeException,Exception,catch,throw)
+import Control.Exception.Safe (Exception,SomeException,catch,throw)
 import Control.Monad.Extra hiding (fromMaybeM)
-import DA.Ledger (LedgerService,PartyDetails(..),Party(..),Token)
-import Data.Aeson
-import Data.Aeson.Text
-import Data.List.Extra as List
+import DA.Ledger (LedgerService,Party(..),PartyDetails(..),Token)
 import qualified DA.Ledger as L
+import Data.Aeson ((.=))
+import qualified Data.Aeson as A
+import Data.Aeson.Text
 import qualified Data.ByteString as BS
+import qualified Data.ByteString.Char8 as BSC
+import Data.List.Extra as List
+import Data.String (IsString)
 import qualified Data.Text as T
 import qualified Data.Text.Lazy as TL
 import qualified Data.Text.Lazy.IO as TL
+import Data.Maybe
+import GHC.Generics
+import Network.HTTP.Simple
 import System.Environment
 import System.Exit
 import System.FilePath
 import System.IO.Extra
 import System.Process.Typed
 
 import qualified DA.Daml.LF.Ast as LF
-import DA.Daml.Compiler.Fetch (LedgerArgs(..),runWithLedgerArgs, fetchDar)
+import DA.Daml.Compiler.Fetch (LedgerArgs(..),runWithLedgerArgs,fetchDar)
 
 import DA.Daml.Helper.Util
 import DA.Daml.Project.Util (fromMaybeM)
 
+data LedgerApi
+  = Grpc
+  | HttpJson
+  deriving (Show, Eq)
+
 data LedgerFlags = LedgerFlags
```

Comment:
Do you know why we have this and `LedgerArgs`? That seems a bit redundant. It's also strange that import the compiler here. This is completely unrelated but might need some cleanup in the future. In the meantime, I think it would make sense to add the `api` field also do `LedgerArgs` instead of passing it around individually.

Reply:
I think the `LedgerArgs/LedgerFlags` and how the gRPC api is called would need some cleanup, it seems unnecessarily complicated to me.

Reply:
Looking at it, I think the module `Compiler.Fetch` should move to the helper, and the `LedgerArgs` should become `LedgerFlags` or vice versa. I'd rather not add an unused field to `LedgerArgs` and clean this up in the near future.